Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/modules/diagnosiskeyprovider/DefaultDiagnosisKeyProvider.kt

```diff
@@ -0,0 +1,110 @@
+@file:Suppress("DEPRECATION")
+
+package de.rki.coronawarnapp.nearby.modules.diagnosiskeyprovider
+
+import com.google.android.gms.nearby.exposurenotification.ExposureConfiguration
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
+import de.rki.coronawarnapp.util.GoogleAPIVersion
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Singleton
+class DefaultDiagnosisKeyProvider @Inject constructor(
+    private val googleAPIVersion: GoogleAPIVersion,
+    private val submissionQuota: SubmissionQuota,
+    private val enfClient: ExposureNotificationClient
+) : DiagnosisKeyProvider {
+
+    override suspend fun provideDiagnosisKeys(
+        keyFiles: Collection<File>,
+        configuration: ExposureConfiguration?,
+        token: String
+    ): Boolean {
+        return try {
+            if (keyFiles.isEmpty()) {
+                Timber.tag(TAG).d("No key files submitted, returning early.")
+                return true
+            }
+
+            val usedConfiguration = if (configuration == null) {
```

Comment:
How about 
`val usedConfiguration = configuration ?:  ExposureConfiguration.ExposureConfigurationBuilder().build()`

Reply:
Then we'd still need an `if` check to log a warning that the configuration was null, right? Not much gained in terms of readability.